src/plugins/data/common/search/opensearch_search/types.ts

```diff
@@ -52,6 +52,7 @@ export type ISearchRequestParams<T = Record<string, any>> = {
 export interface IOpenSearchSearchRequest
   extends IOpenSearchDashboardsSearchRequest<ISearchRequestParams> {
   indexType?: string;
+  dataSourceId?: string;
```

Comment:
why do we add in `IOpenSearchSearchRequest` but not in `IOpenSearchDashboardsSearchRequest`?

Reply:
I think you are right, should be in` IOpenSearchDashoardsRequest`. Since it doesn't affect the actual logic, I'll address it along with future PR